onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -1,109 +1,40 @@
-        for filename in os.listdir(logs_directory):
-            if filename.endswith(".txt") and (filename.startswith("labels.txt") == False): 
-                with open(os.path.join(logs_directory, filename), 'r') as file_object:
-                    line = file_object.readline()
-                    spaces_removed_data = []
-                    while line:
-                        # at each line check for a match with a regex
-                        key, match = _parse_line(line)
-                        if key == 'data': 
-                            data = match.group(0)
-                            for i in data.split(','):
-                                try:
-                                    if i != '':
-                                        spaces_removed_data.append(float(i.strip()))
-                                except ValueError:
-                                    print("error on line"+ i)
-
-                            #add min and max values to the dictionary
-                            if (filename.split(".")[0] not in maxvalue.keys()) or (max(spaces_removed_data) > maxvalue[filename.split(".")[0]]):
-                                maxvalue[filename.split(".")[0]] = max(spaces_removed_data)
-                            if (filename.split(".")[0] not in minvalue.keys()) or (min(spaces_removed_data) < minvalue[filename.split(".")[0]]):
-                                minvalue[filename.split(".")[0]] = min(spaces_removed_data)
-                    
-                        line = file_object.readline() #next line in file
-
-                    file_object.close() #close the file object
-    
-    #clean up the log files to repeat steps for the next image
-    cleanup_images_from_debug_op_run()
-
-    #print(json.dumps(maxvalue, indent = 4)) #dump the max dictionary
-    #print(json.dumps(minvalue, indent = 4)) #dump the min dictionary
-    for conv_output in maxvalue:
-        scale = (maxvalue[conv_output] - minvalue[conv_output]) / 255 if minvalue[conv_output] !=maxvalue[conv_output] else 1
-        zero_point = round((0 - minvalue[conv_output]) / scale)
-        print(f"{conv_output} : {scale} , {zero_point}")
-
-        # Replace by this
-        # quantized_model = quantize(model, per_channel=False, nbits=8, quantization_mode=QuantizationMode.QLinearOps_Dynamic,
-# 		asymmetric_input_types=False, input_quantization_params=_input_quantization_params,
-# 		output_quantization_params=_output_quantization_params)
+from onnx import helper, TensorProto
+
+def augment_graph(model):
+    '''
+    Adds ReduceMin and ReduceMax nodes to all Conv and MatMul nodes_list in
+    model and ensures their outputs are stored as part of the graph output
+        parameter model: loaded ONNX model (not yet quantized)
+        return: augmented ONNX model
+    '''
+    added_nodes = []
+    added_outputs = []
+
+    for node in model.graph.node:
+        if (node.op_type == "Conv") or (node.op_type == "MatMul"):
```

Comment:
instead of hard coding this here ... maintain a global list ... so that when we add more nodes this can be easily extensible

Reply:
Done!